### PR TITLE
Adjust custom rules for Storage Proxy so you can not pass .source or .sink

### PR DESCRIPTION
**Pull request by zmhassan**
# Description
Make sure that api users can not just pass es.source or es.sink. But rather use "es" for the source or sink.

### CODE REVIEW

**Comment by 4n4nd** [comment]
@zmhassan whats the difference between `localdir` and `local` storage types?

**Comment by zmhassan** [comment]
localdir will read through a directory of logs while local only reads one log file.

**Comment by 4n4nd** [comment]
LGTM but it's a big PR, I think we need one more approval. @HumairAK Can you review this as well?

File: docs/source/storage.rst [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+.. code-block:: shell
+
+   STORAGE_DATASOURCE: “es”
+   STORAGE_DATASINK: “kafka”
+
+.. image:: ../../imgs/storage-example.png
+
+.. DANGER::
+
+   We have deprecated es.source/es.sink and you need to specify "es" instead. See table above.
```
Comment (HumairAK) [inline docs/source/storage.rst]: Deprecated would suggest *.source/*.sink is still part of the software, in this PR it seems that that it has been removed entirely, since now we get an exception for this. So I'd suggest rewording to mention it's removed entirely, or just take this bit out imo.

File: anomaly_detector/storage/storage_catalog.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 
 class StorageCatalog(object):
-    """Storage Catalog for storing data to and from ML Runs."""
+    """Storage Catalog should not be used directly. You should utilize storage proxy to access your data pipeline."""
```
Comment (HumairAK) [inline anomaly_detector/storage/storage_catalog.py]: Can we break this sentence up (it's really long atm)

File: anomaly_detector/exception/exceptions.py [hunk 3]
```diff
@@ -39,3 +39,11 @@ class EmptyDataSetException(Exception):
     def __init__(self, msg="Empty dataset returned from ingest"):
         """Initialize message."""
         self.message = msg
+
+
+class InvalidStorageProxyConfigException(Exception):
+    """Validates there is empty dataset returned from ingest."""
```
Comment (HumairAK) [inline anomaly_detector/exception/exceptions.py]: Typos here, suggestion on rewording: Validate that there was an empty dataset returned from ingest.

File: anomaly_detector/storage/storage_proxy.py [hunk 4]
```diff
@@ -3,15 +3,23 @@
...
     """Storage Proxy for facilitating communication with backend."""
 
+    SUFFIX_SOURCE = ".source"
+    SUFFIX_SINK = ".sink"
+
     def __init__(self, config: Configuration):
         """Create storage data source and sinks to talk to storage backend."""
-        self.source = StorageCatalog(config=config, storage_api=config.STORAGE_DATASOURCE).get_storage_api()
-        self.sink = StorageCatalog(config=config, storage_api=config.STORAGE_DATASINK).get_storage_api()
+        if config.STORAGE_DATASOURCE.endswith(self.SUFFIX_SOURCE) or config.STORAGE_DATASINK.endswith(self.SUFFIX_SINK):
```
Comment (HumairAK) [inline anomaly_detector/storage/storage_proxy.py]: Just wondering - why test for only ".sink" ".source" ? A user could put in ".foo" or ".bar" and pass this right? Would it not be better to check if it matches the existing supported source/sinks instead? e.g. something like "config.STORAGE_DATASINK in ['es', 'local', ... ec.] ?

File: anomaly_detector/exception/exceptions.py (hunk 3 above)
Comment (zmhassan) [reply anomaly_detector/exception/exceptions.py]: I've removed this internally.

File: anomaly_detector/storage/storage_proxy.py (hunk 4 above)
Comment (zmhassan) [reply anomaly_detector/storage/storage_proxy.py]: We are already checking lower down in the stack. So I'll remove this.

File: docs/source/storage.rst (hunk 1 above)
Comment (zmhassan) [reply docs/source/storage.rst]: Fixed
